zcncore/transaction_query_base.go

```diff
@@ -457,6 +406,78 @@ func (tq *TransactionQuery) getFastConfirmation(ctx context.Context, txnHash str
 	return nil, nil, nil, thrown.Throw(ErrTransactionNotFound, strconv.Itoa(result.StatusCode))
 }
 
+func (tq *TransactionQuery) GetInfo(ctx context.Context, query string) (*QueryResult, error) {
+
+	consensuses := make(map[int]int)
+	var maxConsensus int
+	var consensusesResp QueryResult
+	// {host}{query}
+	err := tq.FromAll(ctx, query,
+		func(qr QueryResult) bool {
+			//ignore response if it is network error
+			if qr.StatusCode >= 500 {
+				return false
```

Comment:
We need to collect the failedCount as I did in my fix, otherwise we will still need wait for timeout if some of the sharders are offline that caused the online sharders could not meet the consensus.